src/modules/home/Home.js

```diff
@@ -147,6 +147,30 @@ const Home = (props) => {
     };
   }, [props.currentImageBasemap, customMaps]);
 
+  useEffect(() => {
+    if (props.isImageModalVisible) populateImageSlideShowData();
+    else imageSlideShowData = [];
+  }, [props.isImageModalVisible]);
+
+  const populateImageSlideShowData = () => {
+    toggleHomeDrawerButton();
+    const activeSpotsObj = Object.values(useSpots.getActiveSpotsObj());
+    let image = JSON.parse(JSON.stringify(props.selectedImage));
+    let firstImageID = image.id;
+    let uri = useImages.getLocalImageSrc(image.id);
+    let imageWithURI = {image, uri};
+    imageSlideShowData.push(imageWithURI);
+    let spotsWithImagesList = activeSpotsObj.filter(spot => !isEmpty(spot.properties.images));
+    spotsWithImagesList.map(spot => {
```

Comment:
Should be a reduce. Here's the beginning of the reduce:

    const imagesWithURI = Object.values(useSpots.getActiveSpotsObj()).reduce((acc, spot) => {
        const imagesWithURI1 = spot.properties.images && spot.properties.images.reduce((acc1, image) => {